addon/index.js

```diff
@@ -32,10 +34,80 @@ function maybeUnwrapProxy(o) {
   return isProxy(o) ? maybeUnwrapProxy(safeGet(o, 'content')) : o;
 }
 
+function deepNotifyPropertyChange(changeset, path) {
+  let paths = path.split('.');
+  let maybeDynamicPathToNotify = null,
+    lastPath = paths.pop(),
+    current = changeset,
+    i;
+
+  for (i = 0; i < paths.length; ++i) {
+    if (Object.prototype.hasOwnProperty.call(current, paths[i])) {
+      current = safeGet(current, paths[i]);
+    } else {
+      maybeDynamicPathToNotify = paths[i];
+      break;
+    }
+  }
+  const pathToNotify = maybeDynamicPathToNotify ? maybeDynamicPathToNotify : lastPath;
+  notifyPropertyChange(current, pathToNotify);
```

Comment:
I wonder if we should make `notifyPropertyChange` opt-in in a next major 4.0 release?  Generally, we gave patterns that obfuscate the need for these reactivity systems.

Reply:
Or am I missing a crucial need for `notifyPropertyChange`, even in the new reactivity world?

Reply:
🤔🤔🤔 maybe it's no longer needed, you're right, gonna have a look, one think I still need to look for is reactivity around arrays, i.e. computeds which use `changeset.someArray.@each.value`